bucket_writer.go

```diff
@@ -0,0 +1,32 @@
+package bbolt
+
+import (
+	"go.etcd.io/bbolt/internal/common"
+)
+
+func fetchBucketsPages(bk *Bucket) []*common.Page {
```

Comment:
@ahrtr  i wanted to know what do u think abt this solution ? 

I know i can use 

- `pageNode()` https://github.com/etcd-io/bbolt/blob/19a4434f680c10a9653eb4d30e47c6bf65071a32/bucket.go#L821
- tx.checkBucket()`https://github.com/etcd-io/bbolt/blob/f7d0ed5185290b60973dae3d6f9cac123bb26547/tx_check.go#L74


there are also other ways .. i just delved more into the code and tried to find my way .. please let me know 👍🏽

Reply:
The `tx.check` uses some un-exported methods/functions (i.e `forEachPage`, etc.).  It should be better to use exported/public methods/API (i.e. cursor,  ForEach), please read https://github.com/etcd-io/bbolt?tab=readme-ov-file#iterating-over-keys

Reply:
+1, i actually tried to use the `cursor`, but the issue that the exported `First()` that I can use, returns `nil` for `inlineBuckets`

https://github.com/etcd-io/bbolt/blob/7a957f94b271666d839ab867ebb98eaa95a5d047/cursor.go#L38C1-L40


iiuc, here is the algorithm to verify the buckets content 

- fetch the bucket's rootPageId, which is `leafPage`
- add the rootPageId to the result
- traverse the rootPage'c contents ( i.e. leafPageElements)
- if the key contains subBucket 
- - if it is: `inlinedBucked`, where the bucket's `rootPageId == 0`, then add the inlineBucket's content to the result
- - else: the value contains subBucket header only, I fetch the subBucket's rootPageId and recursively traverse

please correct me if i am wrong

Reply:
Returning nil (value) means that it's a bucket.

Suggest to read 
- The doc on iterating over keys: https://github.com/etcd-io/bbolt?tab=readme-ov-file#iterating-over-keys
- How bbolt processes compact command. It iterates on each key & buckets in the db. Now you only need to iterates on a single bucket. https://github.com/etcd-io/bbolt/blob/086cd74c573fa130bd4963c60326fed7a6a1a446/compact.go#L8

Could you please focus on this to get it resolved?

Reply:
sure 👍🏽 

will finish it today 👍🏽